### ISSUE TITLE
Toggleable & spawnable terminal window buffer 

### ISSUE THREAD

**Issue by siduck** [NvChad/nvterm#6]

**Comment by siduck** [comment]
Spawnable term window could be like : 
```
:execute 'terminal' | let b:term_type = 'wind' | startinsert <CR>"
```

**Comment by zbirenbaum** [comment]
this is functionally equivalent to just doing `:term`

**Comment by siduck** [comment]
yes ik its :term, but how to toggle it? :thinking: 

![image](https://user-images.githubusercontent.com/59060246/167980261-c1d2d5b8-92f2-479d-a58c-42b2867601f3.png)

**Comment by zbirenbaum** [comment]
This is because it fundamentally is not how windows function. Buffers are not windows.

I know how to implement this, but both can't until next week, and am not sure I should given that functionally there's no difference between this and literally doing `C-z` running your command in the normal term, and then `fg` to resume.

In fact, doing suspend and resume strategy is actually much better since native term is much faster than neovim emultor.

**Comment by siduck** [comment]
C-z exits nvim 

![image](https://user-images.githubusercontent.com/59060246/171466018-e6ad49c3-1409-4173-a209-69b9983f4785.png)

**Comment by faraazahmad** [comment]
Is this issue related to functionality where `toggle` opens a new terminal if not already open?

**Comment by siduck** [comment]
> Is this issue related to functionality where `toggle` opens a new terminal if not already open?

no, this is for toggling new terminal window ( listed ) if thats possible @zbirenbaum

**Comment by zbirenbaum** [comment]
> > Is this issue related to functionality where `toggle` opens a new terminal if not already open?
> 
> no, this is for toggling new terminal window ( listed ) if thats possible @zbirenbaum

Again, what you are referring to here as a window is actually a buffer. The implementation of the behavior you are describing would be to replace the buffer in the current window with a terminal and then restore it on toggle. I just took some time to see how it would have to be implemented and came across a few problems.

1. This leads to all sorts of messy stuff to save and restore the window options since things like eolchars, linenumbers, etc will all have to be changed when the terminal is opened, and then restored when it is closed. 
2. If the user does anything to close the terminal other than use the dedicated toggle function, it would make a mess of everything and cause tons of bugs.
3. Due to 1 & 2, I would then have to write autocmd's and do a ton of state tracking that is currently unnecessary
4. The only people who could feasibly have a use for this are those who use bufferline and plugins like it. 

This would make much more sense if it opened in a new tabpage, which would actually provide the behavior you want and have none of the above issues, but would not integrate with the plugins that treat buffers as tabs. For that reason, while I am happy to implement this with a tabpage, 

I don't want to do this the way you are thinking about it, as it 1. encourages bad habit 2. Exacerbates confusion between what a buffer and a window actually are 3. Is about 10x more work than quite literally any other feature would require to implement due to the debugging I foresee it requiring.

It isn't undoable, so if it would really mean a lot to you to have this feature, I am willing to put in that effort to write it at some point, but I am going to have to give priority to some other features I am working on in other repos at the moment.

If you want the tabpage implementation, it would be extremely quick to write, and I can do it immediately.

@siduck Let me know your thoughts, and please read this carefully, as all of the above info is relevant.
